### PR TITLE
AWS opensearch docs

### PR DESCRIPTION
**Pull request by himsharma01**
## Purpose of this pull request

This pull request... 

## Select the type of change

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions, updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

### CODE REVIEW

**Comment by himsharma01** [comment]
This PR is in-progress, expect more changes on the way.

**Comment by kimsauce** [comment]
@himsharma01 , thanks for submitting! I've converted it to Draft mode. Once you're ready, you can click **Ready for review**.

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 1]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
 <img src={useBaseUrl('img/integrations/amazon-aws/amazon-opensearch-service-logo.png')} alt="Thumbnail icon" width="50"/>
 
-Amazon OpenSearch Service is a managed service that makes it easy to deploy, operate, and scale OpenSearch clusters in the AWS Cloud. Amazon OpenSearch Service supports OpenSearch and legacy Elasticsearch OSS (up to 7.10, the final open source version of the software). When you create a cluster, you have the option of which search engine to use. For more details, refer to the [AWS documentation](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/what-is.html).
+Amazon OpenSearch Service is a managed service that makes it easy to deploy, operate, and scale OpenSearch clusters in the AWS Cloud. An OpenSearch Service domain is synonymous with an OpenSearch cluster. Domains are clusters with the settings, instance types, instance counts, and storage resources that you specify.
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:11]:
```suggestion
Amazon OpenSearch Service is a managed service that makes it easy to deploy, operate, and scale OpenSearch clusters in the AWS Cloud. An OpenSearch Service domain is synonymous with an OpenSearch cluster, with the settings, instance types, instance counts, and storage resources that you specify. 
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 2]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+Amazon OpenSearch Service is a managed service that makes it easy to deploy, operate, and scale OpenSearch clusters in the AWS Cloud. An OpenSearch Service domain is synonymous with an OpenSearch cluster. Domains are clusters with the settings, instance types, instance counts, and storage resources that you specify. 
 
-## Log and metric types
-* [CloudWatch Metrics](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudwatchmetrics.html)
-* [CloudWatch Logs](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/createdomain-configure-slow-logs.html)
-* [CloudTrail Logs](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudtrailauditing.html)
+The Sumo Logic app for Amazon OpenSearch collects CloudWatch logs, CloudWatch metrics and CloudTrail logs, provides a unified logs and metrics app that provides insights into the operations and utilization of your OpenSearch service. The preconfigured dashboards help you monitor the key metrics by domain names and nodes, view the OpenSearch events for activities, and help you plan the capacity of your OpenSearch service.
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:13]:
```suggestion
- CloudWatch logs
- CloudWatch metrics 
- CloudTrail logs
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 3]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
 
-## Setup
-You can collect the logs and metrics for Sumo Logic's Amazon OpenSearch Service integration by following the below steps.
+## **Log and Metrics types[​](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/monitoring.html)**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:15]:
```suggestion
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 4]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+* OpenSearch using AWS CloudTrail. For details, see [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudtrailauditing.html).
 
-* Collect [AWS CloudTrail Logs](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudtrailauditing.html) using [AWS CloudTrail](/docs/send-data/hosted-collectors/amazon-aws/aws-cloudtrail-source/) source. Amazon OpenSearch Service integrates with AWS CloudTrail, a service that provides a record of actions taken by a user, role, or an AWS service in OpenSearch Service. CloudTrail captures all configuration API calls for OpenSearch Service as events. The captured calls include calls from the OpenSearch Service console, AWS CLI, or an AWS SDK. If you create a trail, you can enable continuous delivery of CloudTrail events to an Amazon S3 bucket, including events for OpenSearch Service. Using the information collected by CloudTrail, you can determine the request that was made to OpenSearch Service, the IP address from which the request was made, who made the request, when it was made, and additional details.
\ No newline at end of file
+### **Sample cloudTrail log messages**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:23]:
```suggestion
### Sample CloudTrail log messages
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 5]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+```
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:26]:
```suggestion
```json

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 6]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+      "domainId": "123456789012/test-domain",
+      "deleted": false,
+      "domainName": "test-domain",
+      "accessPolicies": {...},
+  "requestID": "12345678-1234-1234-1234-987654321098",
+  "eventID": "87654321-4321-4321-4321-987654321098",
+  "eventType": "AwsApiCall",
+  "recipientAccountId": "123456789012"
+}
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:92]:
```suggestion
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 7]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+```
+
+### **Sample queries**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:95]:
```suggestion
### Sample queries
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 8]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+Successful Events by Event Name
+
+```
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:99]:
```suggestion
```sql title="Successful Events by Event Name"
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 9]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+  "eventID": "87654321-4321-4321-4321-987654321098",
+  "eventType": "AwsApiCall",
+  "recipientAccountId": "123456789012"
+}
+
+```
+
+### **Sample queries**
+
+Successful Events by Event Name
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:97]:
```suggestion
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 10]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+```
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:100]:
```suggestion
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 11]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+| if (isEmpty(error_code), "Success", "Failure") as event_status
+| if (isEmpty(username), user, username) as user
+| count as event_count by event_name
+| sort by event_count, event_name asc
+```
+
+Write Latency by Domain Name (Metrics-based)
+
+```
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:119]:
```suggestion
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 12]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+| where (tolowercase(domainname) matches tolowercase("{{domainname}}")) or isBlank(domainname)
+| if (isEmpty(error_code), "Success", "Failure") as event_status
+| if (isEmpty(username), user, username) as user
+| count as event_count by event_name
+| sort by event_count, event_name asc
+```
+
+Write Latency by Domain Name (Metrics-based)
+
+```
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:118]:
```suggestion
```sql title="Write Latency by Domain Name (Metrics-based)"
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 13]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+| json field=requestParameters "domainName" as domainname nodrop
+| if (isBlank(accountid), recipient_account_id, accountid) as accountid
+| where (tolowercase(domainname) matches tolowercase("{{domainname}}")) or isBlank(domainname)
+| if (isEmpty(error_code), "Success", "Failure") as event_status
+| if (isEmpty(username), user, username) as user
+| count as event_count by event_name
+| sort by event_count, event_name asc
+```
+
+Write Latency by Domain Name (Metrics-based)
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:116]:
```suggestion
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 14]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+```
+
+account={{account}} region={{region}} namespace=aws/es domainname={{domainname}} !nodeid=* metric=WriteLatency statistic = average | avg by domainname 
+```
+
+## **Collecting logs and metrics for the Amazon OpenSearch app**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:123]:
```suggestion
## Collecting logs and metrics for the Amazon OpenSearch app
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 15]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+### **Collecting Metrics for Amazon OpenSearch**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:125]:
```suggestion
### Collecting Metrics for Amazon OpenSearch
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 16]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+1. Configure a [Hosted Collector](https://help.sumologic.com/docs/send-data/hosted-collectors/configure-hosted-collector/).
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:127]: Can you please the steps from this document: https://help.sumologic.com/docs/integrations/amazon-aws/dynamodb/#collect-metrics-for-amazon-dynamodb

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 17]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+### **Collecting Metrics for Amazon OpenSearch**
+
+1. Configure a [Hosted Collector](https://help.sumologic.com/docs/send-data/hosted-collectors/configure-hosted-collector/).  
+2. Configure an [Amazon CloudWatch Source for Metrics](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/amazon-cloudwatch-source-metrics/) or [AWS Kinesis Firehose for Metrics Source](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-metrics-source/) (Recommended).  
+3. Namespaces. Select aws/es.  
+4. Metadata. Add an account field to the source and assign it a value that is a friendly name/alias to your AWS account from which you are collecting metrics. The account field allows you to query metrics.  
+   ![Metadata][image2]  
+5. Click Save.
+
+### **Collecting Amazon OpenSearch Events using CloudTrail**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:134]:
```suggestion
### Collecting Amazon OpenSearch CloudTrail Logs
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 18]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+1. Add an [AWS CloudTrail Source](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-cloudtrail-source/) to your Hosted Collector.
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:136]: Can you add the steps similar to the steps in this doc: https://help.sumologic.com/docs/integrations/amazon-aws/dynamodb/#collect-amazon-dynamodb-cloudtrail-logs

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 19]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+     ![Fields][image3]  
+   * Access Key ID and Secret Access Key. Enter your Amazon [Access Key ID and Secret Access Key](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html). Learn how to use Role-based access to AWS [here](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-sources/).  
+   * Log File Discovery \-\> Scan Interval. Use the default of 5 minutes. Alternately, enter the frequency. Sumo Logic will scan your S3 bucket for new data. Learn how to configure Log File Discovery [here](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-sources/).  
+   * Enable Timestamp Parsing. Select the Extract timestamp information from log file entries check box.  
+   * Time Zone. Select Ignore time zone from the log file and instead use, and select UTC from the dropdown.  
+   * Timestamp Format. Select Automatically detect the format.  
+   * Enable Multiline Processing. Select the Detect messages spanning multiple lines check box, and select Infer Boundaries.  
+2. Click Save.
+
+### **Field in Field Schema**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:155]:
```suggestion
### Field in Field Schema
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 20]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+1. [Classic UI](https://help.sumologic.com/docs/get-started/sumo-logic-ui-classic/). In the main Sumo Logic menu, select Manage Data \> Logs \> Fields.  
+   [New UI](https://help.sumologic.com/docs/get-started/sumo-logic-ui/). In the top menu select Configuration, and then under Logs select Fields. You can also click the Go To... menu at the top of the screen and select Fields.  
+2. Search for the `"domainname"` field.  
+3. If not present, create it. Learn how to create and manage fields [here](https://help.sumologic.com/docs/manage/fields/#manage-fields).
+
+### **Field Extraction Rule(s)**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:162]:
```suggestion
### Field Extraction Rule(s)
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 21]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+| parse field=arn ":assumed-role/*" as user nodrop 
+| parse field=arn "arn:aws:iam::*:*" as accountid, user nodrop 
+| json field=requestParameters "domainName" as domainname 
+| if (isBlank(accountid), recipient_account_id, accountid) as accountid 
+| toLowerCase(domainname) as domainname 
+| "aws/es" as namespace 
+| fields region, namespace, domainname, accountid
+```
+
+## **Centralized AWS CloudTrail Log Collection**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:189]:
```suggestion
## Centralized AWS CloudTrail Log Collection
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 22]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+In case, you have a centralized collection of CloudTrail logs and are ingesting them from all accounts into a single Sumo Logic CloudTrail log source, create the following Field Extraction Rule to map a proper AWS account(s) friendly name/alias. Create it if not already present or update it as required.
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:192]:
```suggestion

```
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 23]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+* Rule Name: AWS Accounts  
+* Applied at: Ingest Time  
+* Scope (Specific Data): `_sourceCategory=aws/observability/cloudtrail/logs`  
+* Parse Expression: Enter a parse expression to create an “account” field that maps to the alias you set for each sub account. For example, if you used the “dev” alias for an AWS account with ID "528560886094" and the “prod” alias for an AWS account with ID "567680881046", your parse expression would look like:
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:196]:
```suggestion
```

#### Parse Expression

Enter a parse expression to create an “account” field that maps to the alias you set for each sub account. For example, if you used the `“dev”` alias for an AWS account with ID `"528560886094"` and the `“prod”` alias for an AWS account with ID `"567680881046"`, your parse expression would look like:
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 24]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+| json "recipientAccountId"
+// Manually map your aws account id with the AWS account alias you setup earlier for individual child account
+| "" as account
+| if (recipientAccountId = "528560886094",  "dev", account) as account
+| if (recipientAccountId = "567680881046",  "prod", account) as account
+| fields account
+```
+
+## **Installing the Amazon OpenSearch app**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:208]:
```suggestion
## Installing the Amazon OpenSearch app
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 25]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+Now that you have set up a collection for Amazon OpenSearch, install the Sumo Logic app to use the pre-configured searches and dashboards that provide visibility into your environment for real-time analysis of overall usage.
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:210]:
```suggestion
Now that you have set up a collection for **Amazon OpenSearch**, install the Sumo Logic app to use the [pre-configured dashboards](#viewing-amazon-opensearch-dashboards) that provide visibility into your environment for real-time analysis of overall usage.

import AppInstall from '../../reuse/apps/app-install.md';

<AppInstall/>
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 26]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:211]: Delete from line 211 to 231.

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 27]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+5. Click Next.  
+6. Look for the dialog confirming that your app was installed successfully.  
+   ![app-success.png][image4]
+
+Post-installation
+
+Once your app is installed, it will appear in your Personal folder or the folder that you specified. From here, you can share it with other users in your organization. Dashboard panels will automatically start to fill with data matching the time range query received since you created the panel. Results won't be available immediately, but within about 20 minutes, you'll see completed graphs and maps.
+
+## **Viewing Amazon OpenSearch dashboards**
+
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:234]: import ViewDashboards from '../../reuse/apps/view-dashboards.md';

<ViewDashboards/>

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 28]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+### **Amazon Opensearch \- Overview**
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:235]:
```suggestion
### Amazon Opensearch - Overview
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 29]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+The OpenSearch \- Overview dashboard provides a comprehensive view of the OpenSearch cluster's health, performance, and resource utilization. It offers real-time insights into cluster status, CPU and memory usage, storage metrics, document management, and read/write latencies across different domains.
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:237]:
```suggestion
The **Amazon OpenSearch - Overview** dashboard provides a comprehensive view of the OpenSearch cluster's health, performance, and resource utilization. It offers real-time insights into cluster status, CPU and memory usage, storage metrics, document management, and read/write latencies across different domains.
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md [hunk 30]
```diff
@@ -8,20 +8,345 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
...
+
+Use this dashboard to:
+
+* Monitor the overall health of OpenSearch clusters with color-coded status indicators (green, yellow, red) and quickly identify the number of clusters in each state.  
+* Track resource utilization, including average CPU and JVM memory usage, both overall and by individual domain names.
```
Comment (JV0812) [inline docs/integrations/amazon-aws/amazon-opensearch-service.md:242]:
```suggestion
* Track resource utilization, including average CPU and JVM memory usage, by both overall and individual domain names.  
```

File: docs/integrations/amazon-aws/amazon-opensearch-service.md (hunk 1 above)
Comment (himsharma01) [reply docs/integrations/amazon-aws/amazon-opensearch-service.md:11]: This text is from AWS document which describes relation between domain and cluster. I think we can keep this as it is.

**Review by JV0812 — APPROVED** [review]
LGTM

**Pull request merged**